hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -307,6 +308,7 @@ private MetricDescriptor enhanceDescriptor(MetricDescriptor descriptor, long tim
                             .withIncludedTarget(MANAGEMENT_CENTER)
                             // we add "client" and "timestamp" tags for MC
                             .withTag(METRICS_TAG_CLIENT, getUuid().toString())
+                            .withTag(METRICS_TAG_CLIENTNAME, clientName)
```

Comment:
Shouldn't we make either of `METRICS_TAG_CLIENT` or `METRICS_TAG_CLIENTNAME` a discriminator? Discriminators are used to adding extra info to the metrics definition that it captures the "entity" provided the given metric. It's basically a distinguished tag. It can make processing the metric easier. I think it should be done in the initial implementation with `METRICS_TAG_CLIENT`, but now we can choose between the two mentioned tags. WDYT @emre-aydin?

Reply:
I didn't know about the discriminator tag. In this case, I think making the UUID is a better choice. AFAIK clients can have the same name.

Reply:
A small note here. `METRICS_TAG_CLIENTNAME` are not unique and there may be multiple clients with the same name, so `METRICS_TAG_CLIENT` should be used as a discriminator.

Reply:
Hello, I tried it out and in the prometheus exporter we don't print the discriminator key-value as a metric label. So we can

a) set the clientUuid both as a tag and a discriminator, not sure if such duplication is ok
b) set the clientUuid only as a discriminator, but not as a regular tag - it would be a BC-breaking change, and also the UUID wouldn't be usable by Prometheus clients anymore

One more concern: as far as I understand this `#enchanceDescriptor()` method adds additional tags to an already existing metric, sent from the client. So what happens if the client already set a discriminator for the metric? Isn't it a problem that we override it here, by using the clientUuid as the discriminator? It doesn't look like a future-proof approach for me, so maybe it is better to leave this as-is and not populate the discriminator? WDYT?

Reply:
> Hello, I tried it out and in the prometheus exporter we don't print the discriminator key-value as a metric label. So we can

So, it totally ignores the discriminator? If so, it sounds wrong, as other parts of treat discriminator as a tag.

Reply:
As for the collision problem, I agree that it might be a problem. Even if clients don't use discriminator now, nothing stops them from doing that in future.

Reply:
You're right with the potential override. We do set the discriminator on the client side btw. For example, we set the data structures names in that field. This change is good as is then, sorry for the noise.

On the other hand, ignoring the discriminator doesn't sound good indeed. How a map metric looks like then? If MC converts the discriminator to regular tags, it should be present as a tag `name=<ds_name>` in the Prometheus export.

Reply:
> So, it totally ignores the discriminator? If so, it sounds wrong, as other parts of MC treat discriminator as a tag.

I've checked Prometheus exporter impl and it consumes MC's internal data point representation, which has the discriminator merged into tags. So, we should be safe here.

Reply:
Re-tested, we *do* expose discriminators as tags in Prometheus metrics ([the discriminator is added to the tag list](https://github.com/hazelcast/management-center/blob/master/src/main/java/com/hazelcast/webmonitor/metrics/imdg/ImdgMetricDataPointBuilder.java#L79) internally in MC).

Reply:
Thanks for confirming this @erosb @puzpuzpuz :+1: